// @mft/moneyhub-api-client/dist/request.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const got_1 = __importDefault(require("got"));
const query_string_1 = __importDefault(require("query-string"));
const R = __importStar(require("ramda"));
const getResponseBody = (err) => {
    let body = {};
    try {
        const { code, message, details } = JSON.parse(R.pathOr("{}", ["response", "body"], err));
        body = { code, message, details: typeof details === "object" ? JSON.stringify(details) : details };
    }
    catch (e) {
        body = {};
    }
    return body;
};
const attachErrorDetails = (err) => {
    const { code, message, details } = getResponseBody(err);
    err.error = code;
    err.error_description = message;
    err.error_details = details;
    throw err;
};
exports.default = ({ client, options: { timeout }, }) => async (url, opts = {}) => {
    const gotOpts = {
        method: opts.method || "GET",
        headers: opts.headers || {},
        searchParams: query_string_1.default.stringify(opts.searchParams),
        timeout,
    };
    if (opts.cc) {
        const { access_token } = await client.grant({
            grant_type: "client_credentials",
            scope: opts.cc.scope,
            sub: opts.cc.sub,
        });
        gotOpts.headers.Authorization = `Bearer ${access_token}`;
    }
    if (opts.body) {
        gotOpts.json = opts.body;
    }
    if (opts.form) {
        gotOpts.form = opts.form;
    }
    if (opts.formData) {
        gotOpts.body = opts.formData;
    }
    const req = (0, got_1.default)(url, gotOpts);
    if (opts.returnStatus) {
        return req.then((res) => res.statusCode)
            .catch(attachErrorDetails);
    }
    return req.json()
        .catch(attachErrorDetails);
};
