// @mft/moneyhub-api-client/dist/requests/transaction-files.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const form_data_1 = __importDefault(require("form-data"));
exports.default = ({ config, request }) => {
    const { resourceServerUrl } = config;
    return {
        addFileToTransaction: async ({ userId, transactionId, fileData, fileName }) => {
            const formData = new form_data_1.default();
            formData.append("file", fileData, fileName);
            return request(`${resourceServerUrl}/transactions/${transactionId}/files`, {
                method: "POST",
                formData,
                cc: {
                    scope: "transactions:read:all transactions:write",
                    sub: userId,
                },
            });
        },
        getTransactionFiles: async ({ userId, transactionId }) => request(`${resourceServerUrl}/transactions/${transactionId}/files`, {
            cc: {
                scope: "transactions:read:all transactions:write",
                sub: userId,
            },
        }),
        getTransactionFile: async ({ userId, transactionId, fileId }) => request(`${resourceServerUrl}/transactions/${transactionId}/files/${fileId}`, {
            cc: {
                scope: "transactions:read:all transactions:write",
                sub: userId,
            },
        }),
        deleteTransactionFile: async ({ userId, transactionId, fileId }) => request(`${resourceServerUrl}/transactions/${transactionId}/files/${fileId}`, {
            method: "DELETE",
            cc: {
                scope: "transactions:read:all transactions:write",
                sub: userId,
            },
            returnStatus: true,
        }),
    };
};
